server/src/main/java/org/elasticsearch/index/analysis/FieldNameAnalyzer.java

```diff
@@ -48,4 +48,20 @@ protected Analyzer getWrappedAnalyzer(String fieldName) {
         // Fields need to be explicitly added
         throw new IllegalArgumentException("Field [" + fieldName + "] has no associated analyzer");
     }
+
+    public boolean containsBrokenAnalysis(String field) {
```

Comment:
This is moved from the fastvector highlighter - it would be nice to actually fix this in Lucene rather than have this annoying check.